core/src/test/scala/unit/kafka/log/LogCleanerManagerTest.scala

```diff
@@ -236,13 +236,17 @@ class LogCleanerManagerTest extends JUnitSuite with Logging {
       recoveryPoint = 0L,
       scheduler = time.scheduler,
       time = time,
-      brokerTopicStats = new BrokerTopicStats)
+      brokerTopicStats = new BrokerTopicStats,
+      maxProducerIdExpirationMs = 60 * 60 * 1000,
+      producerIdExpirationCheckIntervalMs = 10 * 60 * 1000,
+      logDirFailureChannel = null)
```

Comment:
In this particular case I think it is OK to pass null simply because it is not expected to be used.. if it is then it will throw NullPointerException and we will discover this when test fails.

Anyway, I have updated the patch to replace null with `new LogDirFailureChannel(10)`. It also works for me.